[IV-23-24] Objetivo 3: Automatización de operaciones

johnwaves opened the pull request:
* [x] ¿Incluye el fichero `iv.yaml` las claves de todos los objetivos
anteriores?
* [x] ¿Se ha desarrollado el fichero para las tareas a partir de un
   issue?
* [x] ¿Ese issue está asignado a un milestone?
* [x] ¿El milestone incluye, entre sus productos mínimamente viables, la
  existencia de herramientas que permitan automatizar tareas?
* [x] ¿Hemos justificado correctamente en caso de elegir un gestor de
   dependencias el que se use como gestor de tareas?
* [x] ¿El PR, que también es un issue, está asignado al mismo milestone?
* [x] ¿La tarea `check` efectivamente comprueba la sintaxis de los fuentes
  existentes y sólo eso?
* [x] ¿Se ha comprobado que las instrucciones incluidas en el `README.md` son
completas y correctas?

johnwaves committed:
Añade apartado para OBJ3

johnwaves committed:
Merge pull request #45 from johnwaves/main

Corrige fallos en rama Objetivo-3

johnwaves committed:
Elige gestor de dependencias closes #43

Se añade un archivo donde se explica la elección del gestor de dependencias
que se usará para el desarrollo de este proyecto en base a los criterios
previamente establecidos.

johnwaves committed:
Elige gestor de tareas closes #44

Se añade otro archivo donde se motiva la elección del gestor de tareas para
este proyecto.

johnwaves committed:
Añade archivo de configuración para Task

Se ha creado un archivo para llevar a cabo la configuración del
gestor de tareas. Previamente se ha instalado Task en local. Se
ha añadido también un comando para la comprobación de la
sintaxis de los fuentes del proyecto.

johnwaves committed:
Actualiza claves closes #46

Se ha añadido la clave automatizar al archivo para cumplir con
los requisitos del objetivo. Las subclaves, "fichero" y "orden"
añadidas referencian el archivo que se utilizará para ejecutar
las tareas asociadas al proyecto y el comando necesario para
ello, respectivamente.

johnwaves committed:
Añade comando para compilar proyecto closes #46

El comando que se ha añadido se utilizará para compilar el proyecto.
A su vez, se menciona que el commit 15aa5ce también pertenece al
issue #46.

johnwaves committed:
Actualiza contenido README

El archivo `README` incluye ahora información acerca de las herramientas
que se utilizarán y sobre el uso de las órdenes disponibles.

pluque01 commented on docs/gestor_dependencias.md:
```diff
@@ -0,0 +1,16 @@
+A pesar de la existencia de varias herramientas capaces de gestionar dependencias en proyectos
+con Go, como pueden ser Glide, Dep o Govendor, la elegida para este proyecto es **Go Modules**.
+
+La elección se ha hecho en base a los criterios especificados en este [issue](https://github.com/johnwaves/recambios-express/issues/43), dado que:
```
Es mejor si escribes los criterios en este mismo documento, así los podemos revisar directamente y sugerir algún cambio.

johnwaves replied:
De acuerdo, los añado. Gracias!

pluque01 commented on docs/gestor_dependencias.md:
```diff
@@ -0,0 +1,16 @@
+# Herramienta para la gestión de dependencias
+
+A pesar de la existencia de varias herramientas capaces de gestionar dependencias en proyectos
+con Go, como pueden ser Glide, Dep o Govendor, la elegida para este proyecto es **Go Modules**.
```
Puedes poner enlaces a las webs de las herramientas que mencionas.

pluque01 commented on docs/gestor_tareas.md:
```diff
@@ -0,0 +1,7 @@
+# Herramienta para la gestión de tareas
+
+Como herramienta de construcción, compilación y automatización de tareas se ha elegido **Task**, pues sigue los [criterios](https://github.com/johnwaves/recambios-express/issues/44) especificados.
```
Lo mismo que para el gestor de dependencias, pon los criterios en el documento.

pluque01 commented on docs/gestor_tareas.md:
```diff
@@ -0,0 +1,7 @@
+
+Además del soporte activo por parte de la comunidad y el equipo desarrollador, el uso de este gestor de tareas resulta ser más simple que otros, como es el caso de Make o Mage, dado que está escrito en Go y su sintaxis es sencilla. Su configuración se centra en un archivo con formato `YAML` donde se definen las tareas a realizar. Task también permite definir variables de entorno para las pruebas  ...
+
+Quedan, por tanto, descartados Make y Mage por no ofrecer una sintaxis tan legible como la de Task ni mayor facilidad en la redacción de scripts.
```
No veo que se mencionen las características de Make ni de Mage.

pluque01 commented on Taskfile.yml:
```diff
@@ -0,0 +1,22 @@
+  check:
+    desc: Comprueba la sintaxis del proyecto.
+    cmds:
+     - gofmt -l -s ./pkg/
```
¿Las opciones `-l` y `-s` juntas son por algo en especial? Lo pregunto desde el desconocimiento.

johnwaves replied:
Cuando lo escribí entendí que con `-l` se listaban los archivos que contenían líneas con errores de sintaxis, pero acabo de comprobar que únicamente imprime aquellos archivos que tienen un formato diferente al de `gofmt`. Voy a cambiarlo y documentarlo. Gracias por preguntar porque así me he podido dar cuenta del error 😅.

pluque01 reviewed (commented):
Hola! Te comento un par de cosas. :))

johnwaves committed:
Actualiza información gestor de tareas #44

Añade los criterios de selección para el gestor de taras, así como algunas
características de las demás herramientas descartadas.

pluque01 commented:
Cuando quieras que le vuelva a echar un vistazo mencióname :))
Ánimo!!

johnwaves committed:
Añade enlace a Task #44

johnwaves committed:
Actualiza información gestor de dependencias #43

Añade los criterios de selección para el getor de dependencias y
enlaces a las herramientas descritas pero no seleccionadas.

johnwaves committed:
Modifica la tarea check closes #46

Corrige los comandos de la tarea check. Ahora se utiliza la opción -e para
informar de todos los errores de sintaxis que puede haber en los archivos
del directorio pkg y se redirige hacia la salida estándar mediante /dev/null.

johnwaves commented:
> Cuando quieras que le vuelva a echar un vistazo mencióname :)) Ánimo!!

De acuerdo. Puedes revisarlo cuando quieras, está modificado 😄.

pluque01 reviewed (approved):
Veo que has corregido lo que te comenté. Creo que has hecho un buen trabajo y pienso que está listo para revisión. Ánimo! :))

johnwaves commented:
> Veo que has corregido lo que te comenté. Creo que has hecho un buen trabajo y pienso que está listo para revisión. Ánimo! :))

¡Muchas gracias por tu trabajo! 😉

johnwaves commented:
@JJ listo para revisión.

JJ commented on docs/gestor_dependencias.md:
```diff
@@ -0,0 +1,13 @@
+- manejo de versiones, y
+- soporte e integración con diversas herramientas de construcción.
+
+A pesar de la existencia de varias herramientas que cumplen con estos requisitos, como pueden ser [Glide](https://glide.readthedocs.io/en/latest/), [Dep](https://golang.github.io/dep/docs/introduction.html) o [Govendor](https://pkg.go.dev/github.com/kardianos/govendor), la elegida para este proyecto es [**Go Modules**](https://go.dev/ref/mod#introduction). Este gestor asegura la compatibilidad co ...
```
Pero ¿glide se usa todavía?

johnwaves replied:
Cuando hice la investigación vi en foros que hasta el año pasado aún había gente que lo utilizaba 🥴.

JJ reviewed (approved):
:+1:
